tools/reviewer/webapp/src/app/dashboard/file-changes/services/state.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import {
+  BranchInfo,
+  Diff,
+  File,
+  TextDiff,
+  Thread,
+} from '@/shared/proto';
+
+// The method keeps state of file-changes
+@Injectable()
+export class StateService {
+  isLoading: boolean;
+  diff: Diff;
+  localThreads: Thread[];
+  branchInfo: BranchInfo;
+  textDiff: TextDiff;
+  diffId: string;
+  language: string;
+  file = new File();
+  isCommitFound: boolean;
+  leftCommitId: string;
+  rightCommitId: string;
+  commitIdList: string[] = [];
+  subscription = new Subscription();
+
+  createFile(): File {
+    const file: File = new File();
+    file.setFilename(this.file.getFilename());
+    file.setRepoId(this.branchInfo.getRepoId());
+    file.setWorkspace(this.diff.getWorkspace());
+
+    return file;
+  }
+
+  getCommitBlockList(): string[] {
```

Comment:
What is a commit block list?
Does it always have 2 elements?
If it's to hold the 2 commits in one variable, maybe it would be clearer to use a class or a proto.

Reply:
Yeah, it's not clear part. Thing is we have `blockIndex`, which may be 0 or 1.
0 - left, 1 - right.
So we create the "Block List" to be able to get commit by the `blockIndex`